src/Catalog/Helpers/AuditRecordHelpers.cs

```diff
@@ -0,0 +1,196 @@
+            {
+                try
+                {
+                    return new DeletionAuditEntry(uri, await auditingStorage.LoadString(uri, cancellationToken));
+                }
+                catch (JsonReaderException)
+                {
+                    logger.LogWarning("Audit record at {AuditRecordUri} contains invalid JSON.", uri);
+                }
+                catch (NullReferenceException)
+                {
+                    logger.LogWarning("Audit record at {AuditRecordUri} does not contain required JSON properties to perform a package delete.", uri);
+                }
+                catch (ArgumentException)
+                {
+                    logger.LogWarning("Audit record at {AuditRecordUri} has no contents.", uri);
+                }
+
+                return null;
+            }
+
+            public DeletionAuditEntry(Uri uri, string contents)
+            {
+                if (string.IsNullOrEmpty(contents))
+                {
+                    throw new ArgumentException($"{nameof(contents)} must not be null or empty!", nameof(contents));
+                }
+
+                Uri = uri;
+                Record = JObject.Parse(contents);
+                InitValues();
+            }
+
+            /// <summary>
+            /// The <see cref="Uri"/> for the audit entry.
+            /// </summary>
+            public Uri Uri { get; set; }
+
+            /// <summary>
+            /// The entire contents of the audit entry file.
+            /// </summary>
+            public JObject Record { get; set; }
+
+            /// <summary>
+            /// The id of the package being audited.
+            /// </summary>
+            public string PackageId { get; set; }
+
+            /// <summary>
+            /// The version of the package being audited.
+            /// </summary>
+            public string PackageVersion { get; set; }
+
+            /// <summary>
+            /// The <see cref="DateTime"/> the package was deleted.
+            /// </summary>
+            public DateTime? TimestampUtc { get; set; }
+
+            private const string RecordPart = "Record";
+            private const string ActorPart = "Actor";
+
+            private JObject GetPart(string partName)
+            {
+                return (JObject)Record?.GetValue(partName, StringComparison.OrdinalIgnoreCase);
+            }
+
+            private void InitValues()
+            {
+                PackageId = GetPart(RecordPart).GetValue("Id", StringComparison.OrdinalIgnoreCase).ToString();
+                PackageVersion = GetPart(RecordPart).GetValue("Version", StringComparison.OrdinalIgnoreCase).ToString();
+                TimestampUtc =
+                    GetPart(ActorPart).GetValue("TimestampUtc", StringComparison.OrdinalIgnoreCase).Value<DateTime>();
+            }
+        }
+
+        /// <summary>
+        /// Fetches <see cref="DeletionAuditEntry"/>s based on a filter.
+        /// </summary>
```

Comment:
The filter definition is internal. The dev consuming the API will not understand what the filter is about. Maybe instead of "based on a filter" add what the filter does?